packages/kbn-i18n/src/loader.ts

```diff
@@ -137,9 +137,13 @@ export async function getTranslationsByLocale(locale: string): Promise<PlainMess
 
   return files.length
     ? files.reduce(
-        (messages, file) => ({
+        (messages: { locale: string; messages?: Translation['messages'] }, file) => ({
           ...messages,
```

Comment:
__nit:__ same advise here as in the `i18n.ts`, let's be more explicit what we're merging here, we have just 3 properties to merge and define one of them explicitly already anyway (`messages`).